### PR TITLE
Expand documentation for IntoInnerError

### PR DESCRIPTION
**Pull request by steveklabnik**
Mostly adding examples.

r? @alexcrichton

### CODE REVIEW

File: src/libstd/io/buffered.rs [hunk 1]
```diff
@@ -269,6 +269,27 @@ pub struct BufWriter<W: Write> {
...
+/// let mut stream = BufWriter::new(TcpStream::connect("127.0.0.1:34254").unwrap());
+///
+/// // do stuff with the stream
+///
+/// // we want to get our `TcpStream` back, so let's try:
+///
+/// let stream = match stream.into_inner() {
+///     Ok(s) => s,
+///     Err(e) => {
+///         // Here, e is a IntoInnerError
```
Comment (alexcrichton) [inline src/libstd/io/buffered.rs]: s/a/an/

Comment (alexcrichton) [inline src/libstd/io/buffered.rs]: (same below as well)

File: src/libstd/io/buffered.rs [hunk 2]
```diff
@@ -384,9 +405,34 @@ impl<W: Write> Drop for BufWriter<W> {
...
+    ///
+    /// let stream = match stream.into_inner() {
+    ///     Ok(s) => s,
+    ///     Err(e) => {
+    ///         // Here, e is a IntoInnerError, let's log the inner error.
+    ///         //
+    ///         // We'll just 'log' to stdout for this example.
+    ///         println!("{}", e.error());
+    ///
+    ///         panic!("An unexpected error occurred.");
```
Comment (alexcrichton) [inline src/libstd/io/buffered.rs]: Perhaps the `e.error()` could be used in the panic message here instead of printing out?

**Comment by alexcrichton** [comment]
r=me with a few small nits + travis fixes

File: src/libstd/io/buffered.rs (hunk 2 above)
Comment (steveklabnik) [inline src/libstd/io/buffered.rs]: So, the reason I didn't do that was that then the example is just `unwrap()`...

**Comment by steveklabnik** [comment]
fixed the nits, but curious as to your thoughts on the example :)

File: src/libstd/io/buffered.rs (hunk 2 above)
Comment (alexcrichton) [inline src/libstd/io/buffered.rs]: Ah, right

**Pull request merged**